### DISCUSSION TITLE
Bug: MAIN: Changing model mid conversation to a non-vision model results in a HANG if the conversation contains an image.

### DISCUSSION THREAD

**Discussion by spammenotinoz** [open-webui/open-webui#2635] [General]
# Bug Report

## Description
Bug: MAIN: Changing model mid conversation to a non-vision model results in a HANG if the conversation contains an image.

**Bug Summary:** / **Steps to Reproduce:**
1. Start a conversation with a vision capable model, upload an Image and submit message.
2. Upon receipt of response, change model to one that does NOT support models (ie: most)
3. 
MAIN Branch: Hangs for a long time before returning a response..
DEV Brach: Red Error "Model x is not vision capable"

**Expected Behavior:**
Upon Model Change, either Clear entire History, or clear images from History.

**Actual Behavior:**
PROD: Pause for a few minutes before a response.
DEV: Red Error "Model x is not vision capable"

## Environment
- **Open WebUI Version:** [e.g., 0.1.120]
- **Ollama (if applicable):** [N/A

- **Operating System:** [Windows 10 and Ubuntu 20.04]
- **Browser (if applicable):** [Edge and Safari]

## Reproduction Details

**Confirmation:**

- [ x ] I have read and followed all the instructions provided in the README.md.
- [ x ] I am on the latest version of both Open WebUI and Ollama.
- [ N/A ] I have included the browser console logs.
- [ N/A ] I have included the Docker container logs.

## Logs and Screenshots
No Errors in Logs

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
No errors in logs

**Screenshots (if applicable):**
Hang is not applicable in a screen shot, so not included.

## Installation Method
Docker Compose

## Additional Information
N/A

**Comment by blinkyboi** [comment]
Try [unchecking the "Vision" capability](https://github.com/user-attachments/assets/b842b37e-6de4-4a55-a05f-ec4f956bbdae) from the non-vision model? Hope this helps!
